src/components/LongPressable/index.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react'
+import PropType from 'prop-types'
+import Pointable from 'react-pointable'
+
+export default class LongPressable extends React.PureComponent {
+
+  static propTypes = {
+    onLongPress: PropType.func.isRequired,
+    onShortPress: PropType.func,
+    longPressTime: PropType.number,
+    primaryMouseButtonOnly: PropType.bool,
+    dragThreshold: PropType.number,
+    children: PropType.node
+  }
+
+  static defaultProps = {
+    longPressTime: 500,
+    primaryMouseButtonOnly: true,
+    dragThreshold: 0
+  }
+
+  isLongPressing = false
+  dragCounter = 0
+
+  onPointerUp = () => {
+    if (this.timerID) {
+      this.clearLongPressTimer()
+    }
+
+    if (!this.isLongPressing) {
+      this.props.onShortPress()
+    }
+    else {
+      this.isLongPressing = false
+    }
+  }
+
+  onPointerDown = (e) => {
+    if (this.props.primaryMouseButtonOnly) {
+      if (e.pointerType === 'mouse' && e.button !== 0) {
+        return
+      }
+    }
+
+    this.dragCounter = 0
+
+    this.timerID = setTimeout(() => {
+      this.isLongPressing = true
+      this.props.onLongPress()
+    }, this.props.longPressTime)
+  }
+
+  onPointerMove = () => {
+    this.dragCounter++
```

Comment:
My suggestion with the drag threshold was this: clicks can be "sloppy", i.e., the user moves pointer slightly while attempting to click. The intent is to click, not to drag.

With that in mind, rather than cancelling the `shortPress` / `longPress` callbacks if the number of `pointerMove` events exceeds a certain threshold, I think it makes more sense to cancel the callbacks if the *distance* moved exceeds a certain threshold.

Consider that I might accidentally move my mouse back-and-forth or (up-and-down) very slightly, triggering many `pointerMove` events, but not actually moving the pointer very much.

Related: React-beautiful-dnd calls this threshold "sloppyClickThreshold" and has a default of `5px`. I think we should use the same default. (Not zero, as about in `defaultProps`.)

Probably also good to cancel shortPress if dragThreshold is exceeded.

Reply:
How do you like your comments? I feel like I need to document dragThreshold because it's not obvious that it's distance in pixels.

Reply:
I'm not sure. For functions, I've been trying to follow style/tags of [ESDoc](https://esdoc.org/manual/faq.html). But for React components, I'm not sure.

Usually, I think that `propTypes` is enough documentation. In this case, I agree that it might be nice to say slightly more, so how about an inline comment with the particular propType:

```js
static propTypes = {
   ...
   dragThreshold: PropTypes.number.isRequired // distance in pixels
   ...
}
```